lib/puppet/type/sensu_check.rb

```diff
@@ -21,10 +21,13 @@ def should
   def initialize(*args)
     super *args
 
-    self[:notify] = [
-      "Service[sensu-client]",
-      "Service[sensu-server]",
-    ].select { |ref| catalog.resource(ref) }
+    if c = catalog
+      self[:notify] = [
+        "Service[sensu-client]",
+        "Service[sensu-server]",
+        "Service[sensu-enterprise]",
```

Comment:
I expect no problem because`Array#select` filter out the resource ref when absent from the catalog.